openssl-dynamic/src/main/c/tcn_lock_rw.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#include "tcn_lock_rw.h"
+#include <shared_mutex>
+
+tcn_lock_rw_t tcn_lock_rw_new() {
+    auto *m = new std::shared_timed_mutex();
```

Comment:
add a `TODO` that we should use `std::shared_mutex `when we are able to use c++17 (we don't use the timed component).